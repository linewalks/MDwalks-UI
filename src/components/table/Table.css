.table {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.body_b_16 {
  font-weight: bold;
  letter-spacing: -0.5px;
  font-size: 16px;
}

.body_r_18 {
  font-size: 18px;
}

.body_r_20 {
  font-size: 20px;
}

.th {
  padding: 28px 24px;
  text-align: center;
  background: #f2f2f2;
  border-bottom: 2px solid #d4d4d4;
  color: rgba(0, 0, 0, 0.7);
}

.th:first-child {
  border-radius: 10px 0 0 0; 
}

.th:last-child {
  border-radius: 0 10px 0 0;
}

.tr {
  cursor: pointer;
}

.tr:nth-child(even) .td {
  background: #ffffff;
}

.tr:nth-child(odd) .td {
  background: #fafafa;
}

.td {
  text-align: center;
}

.td > a > div, .td > div {
  padding: 24px;
}

.th:first-child, .th:last-child, .td:first-child, .td:last-child {
  white-space: nowrap;
  width: 1%;
}

.th:first-child, .td:first-child {
  padding-left: 50px;
}

.th:last-child, .td:last-child {
  padding-right: 50px;
}

.empty_table_grey_list {
  padding: 80px;
}

.empty_table_grey_list td {
  margin: 0 auto;
  text-align: center;
}

.empty_table_grey_list span {
  margin: auto;
  color: #161616;
  display: block;
  font-size: 20px;
  opacity: 0.6;
}

.empty_table_grey_header {
  font-size: 16px;
  color: rgba(#000000, 0.36);
}