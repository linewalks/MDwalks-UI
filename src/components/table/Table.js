import React from 'react';
import PropTypes from 'prop-types'
import _ from 'lodash'

import THead from '@Components/table/THead';
import TBody from '@Components/table/TBody';
import TFoot from '@Components/table/TFoot';
import styled, { css } from 'styled-components'
import * as variables from '@src/assets/styles/variables'
import * as commonTag from '@Components/common/commonTag'

const sideFit = css`
  tbody {
    .td:first-child, .td:last-child {
      white-space: nowrap;
      width: 1%;
    }

    .td:first-child > a > div,
    .td:first-child > div {
      padding-left: 0
    }

    .td:last-child > a > div,
    .td:last-child > div {
      padding-right: 0
    }

    .td:first-child {
      padding-left: 50px;
    }

    .td:last-child {
      padding-right: 50px;
    }
  }

  thead td {
    &:last-child {
      padding-right: 50px;
    }
  }
  thead th {
    &:first-child, &:last-child {
      white-space: nowrap;
      width: 1%;
    }

    &:first-child {
      padding-left: 50px;
    }

    &:last-child {
      padding-right: 50px;
    }
  }
`

const TableBox = styled.table`
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;

  border-bottom: 1px solid ${variables.colorV1.$grey06};

  ${(props) => (props.notBottom ? 'border-bottom: none' : '')};

  ${(props) => (props.className.split(' ').includes('sideFit') ? sideFit : null)}
`

const WrapScrollTables = styled.div`
  border-bottom: 1px solid ${variables.colorV1.$grey06};
`

const Columns = ({ columns }) => (
  !_.isEmpty(columns) && (
    <colgroup>
      {
        _.map(columns, (width, i) => {
          const key = `columns${i}`
          return (<col key={key} style={{ width }} />)
        })
      }
    </colgroup>
  )
)

Columns.defaultProps = {
  columns: {},
}

Columns.propTypes = {
  columns: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
}

const Table = ({
  data, rowSpanCount, wrapTh, wrapTd, appendRow, className,
  loading, scroll, columns, size,
}) => {
  if (_.isEmpty(data)) {
    return (
      <div>
        There is no data
        <br />
        Please search agains
      </div>
    )
  }

  if (!_.isEmpty(scroll)) {
    return (
      <WrapScrollTables>
        <TableBox className={className} notBottom>
          <Columns columns={columns} />
          <THead
            headers={data.headers}
            subHeaders={data.subHeaders}
            wrapTh={wrapTh}
            loading={loading}
            size={size}
          />
        </TableBox>
        <commonTag.WrapperScrollBars scroll={scroll}>
          <TableBox className={className} notBottom>
            <Columns columns={columns} />
            <TBody
              headers={data.headers}
              subHeaders={data.subHeaders}
              rowData={data.rowData}
              wrapTd={wrapTd}
              appendRow={appendRow}
              rowSpanCount={rowSpanCount}
              size={size}
            />
          </TableBox>
        </commonTag.WrapperScrollBars>
        <TableBox className={className} notBottom>
          <Columns columns={columns} />
          <TFoot footData={data.footData} size={size} />
        </TableBox>
      </WrapScrollTables>
    )
  }

  return (
    <TableBox className={className}>
      <Columns columns={columns} />
      <THead
        headers={data.headers}
        subHeaders={data.subHeaders}
        wrapTh={wrapTh}
        loading={loading}
        size={size}
      />
      <TBody
        headers={data.headers}
        subHeaders={data.subHeaders}
        rowData={data.rowData}
        wrapTd={wrapTd}
        appendRow={appendRow}
        rowSpanCount={rowSpanCount}
        size={size}
      />
      <TFoot footData={data.footData} size={size} />
    </TableBox>
  );
};

Table.defaultProps = {
  data: {},
  rowSpanCount: undefined,
  wrapTh: undefined,
  wrapTd: undefined,
  appendRow: undefined,
  className: '',
  loading: false,
  scroll: {},
  columns: [],
  size: 'medium',
}

Table.propTypes = {
  columns: PropTypes.arrayOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ),
  data: PropTypes.shape({
    headers: PropTypes.arrayOf(
      PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.shape(),
      ]),
    ),
    subHeaders: PropTypes.shape({}),
    rowData: PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.object, PropTypes.array, PropTypes.string, PropTypes.number]),
    ),
    footData: PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.object, PropTypes.array, PropTypes.string, PropTypes.number]),
    ),
  }),
  rowSpanCount: PropTypes.number,
  wrapTh: PropTypes.func,
  wrapTd: PropTypes.func,
  appendRow: PropTypes.func,
  className: PropTypes.string,
  loading: PropTypes.bool,
  scroll: PropTypes.shape({
    y: PropTypes.number,
  }),
  size: PropTypes.string,
}

export default Table
