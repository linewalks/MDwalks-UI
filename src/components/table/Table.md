Table example:

#### List Sub Head Table
```js
import styled from 'styled-components'

const WrapTable = styled.section`
  td, td div { text-align: right; }
  tr td:first-child div { text-align: center; }
`

const show = () => (
  <WrapTable>
    <Table
      data={{
        headers: ['스위칭 의약품', '2013년', '2014년', '성장률'],
        subHeaders: {
          "2013년": ['처방건수', '비율'],
          "2014년": ['처방건수', '비율'],
          "성장률": ['CAGR', 'YOY'],
        },
        group: {
          '유지': ['A'],
          '추가': ['AB', 'AC'],
          '변경': ['B']
        },
        rowData: [
          {
            'a1': 'A',
            'b1': '3,112',
            'b2': '64.7%',
            'c1': '3,474',
            'c2': '66.3%',
            'd1': '12.6%',
            'd2': '33.2%',
          },
        ]
      }}
    />
  </WrapTable>
)

show()
```

#### List Table
```js
<Table
  className={'sideFit'}
  data={{
    headers: ['a', 'b', 'c'],
    rowData: [
      {
        'a': 1,
        'b': 2,
        'c': 3
      },
      {
        'a': 4,
        'b': 5,
        'c': 6
      },
      {
        'a': 7,
        'b': 8,
        'c': 9
      },
    ]
  }}
/>
```

#### Data Array List
```js
<Table
  data={{
    headers: ['a', 'b', 'c'],
    rowData: [
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9]
    ],
    footData: [
      ['t1', 't2', 't3'],
      ['t4', 't5', 't6']
    ]
  }}
/>
```

#### rowSpan List Table
```js
<Table
  data={{
    headers: ['a', 'b', 'c', 'd', 'e'],
    rowData: [
      [{rowSpan: 3, text: 'a1'}, 'b1', {rowSpan: 2, text: 'c1'}, {rowSpan: 2, text: 'd1'}, 'e1'],
      ['b2', 'e2'],
      ['b3', 'c3', 'd3', 'e3'],
      [{rowSpan: 2, text: 'a2', className: 'odd'}, 'b4', 'c4', 'd4', 'e4'],
      ['b5', 'c5', 'd5', 'e5'],
    ]
  }}>
</Table>
```

##### Empty List Table
```js
<Table
  data={{
    headers: ['a', 'b', 'c'],
  }}>
</Table>
```

##### Sort List Table
```js
import React, { useState, useEffect } from 'react'

(() => {
  const [loading, setLoading] = useState(false)
  const onChange = (e) => {
    setLoading(e.target.checked)
  }

  return (
    <>
      <input type="checkbox" onChange={onChange} /><span>loading</span>
      <Table
        size="small"
        loading={loading}
        data={{
          headers: ['a', 'b', { text: 'c', sort: function(a, b) { console.log(a, b) } }],
          rowData: [
            [1, 2, 3],
            [4, 5, 6],
            [7, 8, 9]
          ],
          footData: [
            ['t1', 't2', 't3'],
            ['t4', 't5', 't6']
          ]
        }}
      >
      </Table>
    </>
  )
})()

```

#### scroll Array List
```js
<Table
  columns={[100, 'auto', 300]}
  scroll={{ y: 215 }}
  data={{
    headers: ['a', 'b', 'c'],
    rowData: [
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9],
      [10, 11, 12],
      [13, 14, 15],
    ],
    footData: [
      ['t1',  {colSpan: 2, text: 't3'}],
    ]
  }}
/>
```

#### Table size small
```js
<Table
  data={{
    headers: ['a', 'b', 'c'],
    rowData: [
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9]
    ],
    footData: [
      ['t1', 't2', 't3'],
      ['t4', 't5', 't6']
    ]
  }}
  size='small'  // default: medium
/>
```

#### Row Header Table(Custom)
```js
import styled from 'styled-components'

const WrapTable = styled.section`
  table {
    tr td:first-child {
      background: #f3f6f8;
      border-right: 1px solid #e7ebee;
    }
  }
`

const showTable = () => (
  <WrapTable>
    <Table
      data={{
        headers: [' ', 'b', 'c'],
        rowData: [
          [1, 2, 3],
          [4, 5, 6],
          [7, 8, 9]
        ],
      }}
    />
  </WrapTable>
)

showTable()
```


#### Vertical Scroll Table(Cutom)
```js
import styled from 'styled-components'

const WrapTable = styled.section`
  display: flex;
`

const WrapFixedTable = styled.section`
  width: 200px;
  thead th:last-child {
    border-radius: 0;
  }
  table {
    border-right: 1px solid #e7ebee;
  }
`

const WrapScrollingTable = styled.section`
  overflow-x: scroll;
  thead th:first-child {
    border-radius: 0;
  }
`

const showTable = () => (
  <WrapTable>
    <WrapFixedTable>
      <Table
        data={{
          headers: ['A', 'B'],
          rowData: [
            [{ rowSpan: 2, text: 'A1'}, 'B1'],
            ['B2'],
          ],
        }}
      />
    </WrapFixedTable>
    <WrapScrollingTable>
      <Table
        data={{
          headers: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x'],
          rowData: [
            [1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3],
            [4, 5, 6, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3],
          ],
        }}
      />
    </WrapScrollingTable>
  </WrapTable>
)

showTable()
```