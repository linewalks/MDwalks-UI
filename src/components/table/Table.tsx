import React from 'react';
import _ from 'lodash'
import styled, { css } from 'styled-components'
import { color } from '../../assets/styles/variables'
import * as commonTag from '../common/commonTag'
import THead from './THead'
import TBody from './TBody'
import TFoot from './TFoot'

const sideFit = css`
  tbody {
    .td:first-child, .td:last-child {
      white-space: nowrap;
      width: 1%;
    }

    .td:first-child > a > div,
    .td:first-child > div {
      padding-left: 0
    }

    .td:last-child > a > div,
    .td:last-child > div {
      padding-right: 0
    }

    .td:first-child {
      padding-left: 50px;
    }

    .td:last-child {
      padding-right: 50px;
    }
  }

  thead td {
    &:last-child {
      padding-right: 50px;
    }
  }
  thead th {
    &:first-child, &:last-child {
      white-space: nowrap;
      width: 1%;
    }

    &:first-child {
      padding-left: 50px;
    }

    &:last-child {
      padding-right: 50px;
    }
  }
`

interface ITableBox {
  notBottom?: boolean;
  className?: string;
}

const TableBox = styled.table<ITableBox>`
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;

  border-bottom:   ${(props) => (props.notBottom ? 'none' : `1px solid ${color.$grey06}`)};

  ${(props) => (props.className.split(' ').includes('sideFit') ? sideFit : null)}
`

const WrapScrollTables = styled.div`
  border-bottom: 1px solid ${color.$grey06};
`

interface IColumns {
  columns?: number[];
}

const Columns = ({ columns }:IColumns) => (
  !_.isEmpty(columns) && (
    <colgroup>
      {
        _.map(columns, (width, i) => {
          const key = `columns${i}`
          return (<col key={key} style={{ width }} />)
        })
      }
    </colgroup>
  )
)

Columns.defaultProps = {
  columns: {},
}

interface TableProps extends IColumns {
  data: {
    headers: any[];
    subHeaders?: {
      [key:string]: string[];
    };
    rowData: any[];
    footData: any[];
  };
  wrapTh: () => void;
  wrapTd: (data:any) => React.ReactNode;
  appendRow: () => void;
  className?: string;
  loading: boolean;
  scroll?: {
    y: number;
  };
  placeholder?: React.ReactNode | string;
  sortOrderList?: string[];
  defaultSort?: {
    text: string;
    order: number;
  };
  resetSort?: boolean;
  setResetSort?: () => void;
  size: 'small' | 'medium';
}

const Table = ({
  data, wrapTh, wrapTd, appendRow, className,
  loading, scroll, columns, size, placeholder, defaultSort,
  sortOrderList, resetSort, setResetSort,
}:TableProps) => {
  if (_.isEmpty(data)) {
    return (
      <div>
        There is no data
        <br />
        Please search agains
      </div>
    )
  }

  if (!_.isEmpty(scroll)) {
    return (
      <WrapScrollTables>
        <TableBox className={className} notBottom>
          <Columns columns={columns} />
          <THead
            headers={data.headers}
            subHeaders={data.subHeaders}
            wrapTh={wrapTh}
            loading={loading}
            size={size}
            sortOrderList={sortOrderList}
            defaultSort={defaultSort}
            resetSort={resetSort}
            setResetSort={setResetSort}
          />
        </TableBox>
        <commonTag.WrapperScrollBars scroll={scroll}>
          <TableBox className={className} notBottom>
            <Columns columns={columns} />
            <TBody
              headers={data.headers}
              subHeaders={data.subHeaders}
              rowData={data.rowData}
              wrapTd={wrapTd}
              appendRow={appendRow}
              size={size}
              placeholder={placeholder}
            />
          </TableBox>
        </commonTag.WrapperScrollBars>
        <TableBox className={className} notBottom>
          <Columns columns={columns} />
          <TFoot footData={data.footData} size={size} />
        </TableBox>
      </WrapScrollTables>
    )
  }

  return (
    <TableBox className={className}>
      <Columns columns={columns} />
      <THead
        headers={data.headers}
        subHeaders={data.subHeaders}
        wrapTh={wrapTh}
        loading={loading}
        size={size}
        sortOrderList={sortOrderList}
        defaultSort={defaultSort}
        resetSort={resetSort}
        setResetSort={setResetSort}
      />
      <TBody
        headers={data.headers}
        subHeaders={data.subHeaders}
        rowData={data.rowData}
        wrapTd={wrapTd}
        appendRow={appendRow}
        size={size}
        placeholder={placeholder}
      />
      <TFoot footData={data.footData} size={size} />
    </TableBox>
  );
};

Table.defaultProps = {
  data: {},
  rowSpanCount: undefined,
  wrapTh: undefined,
  wrapTd: undefined,
  appendRow: undefined,
  className: '',
  loading: false,
  scroll: {},
  columns: [],
  size: 'medium',
  placeholder: undefined,
  sortOrderList: ['desc', 'asc', ''],
  defaultSort: {},
  resetSort: false,
  setResetSort: () => {},
}

export default Table
