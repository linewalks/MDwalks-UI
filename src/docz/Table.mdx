---
name: Table
edit: false
---

import { Playground, Props } from 'docz'

import Table from '../components/table/Table'

# card

## SelectedCard

<Playground>
  <div style={{overflow: 'hidden', overflowX: 'scroll'}}>
    <Table
      data={{
        headers: ['a', 'b', 'c'],
        rowData: [
          {
            'a': 1,
            'b': 2,
            'c': 3
          },
          {
            'a': 4,
            'b': 5,
            'c': 6
          },
          {
            'a': 7,
            'b': 8,
            'c': 9
          },
        ]
      }} 
    />
  </div>
</Playground>

## SummaryCard