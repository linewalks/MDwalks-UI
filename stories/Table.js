import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import Table from '@Table/Table';

storiesOf('Table Component', module)
.addDecorator(withKnobs)
.add('basic', () => (
  <Table 
    data={{
      headers: ['a', 'b', 'c'],
      rowData: [
        {
          'a': 1,
          'b': 2,
          'c': 3
        },
        {
          'a': 4,
          'b': 5,
          'c': 6
        },
        {
          'a': 7,
          'b': 8,
          'c': 9
        },
      ]
    }} 
  />
))
.add('with out any rowData', () => (
  <Table 
    data={{
      headers: ['a', 'b', 'c'],
     
    }} 
  />
))