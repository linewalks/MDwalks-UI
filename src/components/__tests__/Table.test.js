import React from 'react';
import { shallow, mount } from 'enzyme';
import Table from '@Table/Table';
import renderer from 'react-test-renderer';

const data = {
  exceptSubHeaders: {
    headers: ['a', 'b', 'c'],
    rowData: [
      {
        'a': 1,
        'b': 2,
        'c': 3
      },
      {
        'a': 4,
        'b': 5,
        'c': 6
      },
      {
        'a': 7,
        'b': 8,
        'c': 9
      },
    ]
  },

  includeSubHeaders: {
    headers: ['a', 'b', 'c'],
    subHeaders: {
      b: ['d', 'e'],
      c: ['f', 'g']
    },
    rowData: [
      {
        'a': 1,
        'd': 2,
        'e': 3,
        'f': 4,
        'g': 5
      },
      {
        'a': 6,
        'd': 7,
        'e': 8,
        'f': 9,
        'g': 10
      },
      {
        'a': 11,
        'd': 12,
        'e': 13,
        'f': 14,
        'g': 15
      },
    ]
  }
}

describe('Table Component', () => {
  it('render error message when there is no data', () => {
    const wrapper = shallow(<Table />)
    expect(wrapper.html()).toEqual("<div>There is no data<br/>Please search agains</div>");
  })

  it('render table when there is data excpet subHeaders', () => {
    const wrapper = renderer.create(<Table data={data.exceptSubHeaders} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  })

  it('render table when there is data include subHeaders', () => {
    const wrapper = renderer.create(<Table data={data.includeSubHeaders} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  })
})
