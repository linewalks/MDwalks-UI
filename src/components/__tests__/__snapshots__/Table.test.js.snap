// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component render table when there is data excpet subHeaders 1`] = `
<table
  className="table"
>
  <thead
    className="body_b_16"
  >
    <tr>
      <th
        className="th"
      >
        A
      </th>
      <th
        className="th"
      >
        B
      </th>
      <th
        className="th"
      >
        C
      </th>
    </tr>
  </thead>
  <tbody
    className={null}
  >
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          1
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          2
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          3
        </div>
      </td>
    </tr>
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          4
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          5
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          6
        </div>
      </td>
    </tr>
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          7
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          8
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          9
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Component render table when there is data include subHeaders 1`] = `
<table
  className="table"
>
  <thead
    className="body_b_16"
  >
    <tr>
      <th
        className="th"
        rowSpan={2}
      >
        a
      </th>
      <th
        className="th"
        colSpan={2}
      >
        b
      </th>
      <th
        className="th"
        colSpan={2}
      >
        c
      </th>
    </tr>
    <tr>
      <td>
        d
      </td>
      <td>
        e
      </td>
      <td>
        f
      </td>
      <td>
        g
      </td>
    </tr>
  </thead>
  <tbody
    className={null}
  >
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          1
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          2
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          3
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          4
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          5
        </div>
      </td>
    </tr>
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          6
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          7
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          8
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          9
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          10
        </div>
      </td>
    </tr>
    <tr
      className="tr body_r_18"
    >
      <td
        className="td"
      >
        <div>
          11
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          12
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          13
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          14
        </div>
      </td>
      <td
        className="td"
      >
        <div>
          15
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
